### PR TITLE
fix for NaN comparisons

### PR DESCRIPTION
**Pull request by thehesiod**
testcase:
```python
import dictdiffer
print(list(dictdiffer.diff([float('nan')], [float('nan')])))

### CODE REVIEW

**Comment by jirikuncar** [comment]
@thehesiod thank you. Can you please include the test-case from the description in https://github.com/inveniosoftware/dictdiffer/blob/master/tests/test_dictdiffer.py so we can better understand which problem your PR solves.

**Comment by thehesiod** [comment]
done!

File: dictdiffer/__init__.py [hunk 1]
```diff
@@ -197,7 +198,8 @@ def check(key):
 
     else:
         # Compare string and integer types and yield `change` flag.
-        if first != second:
+        nan_eq = isinstance(first, float) and math.isnan(first) and isinstance(second, float) and math.isnan(second)
+        if first != second and not nan_eq:
```
Comment (jirikuncar) [inline dictdiffer/__init__.py]: what about something like:
```python
if first != second and not (first != first and second != second):
```

File: tests/test_dictdiffer.py [hunk 2]
```diff
@@ -195,6 +195,11 @@ def test_types(self):
         diffed = next(diff(first, second))
         assert ('change', 'a', (['a'], 'a')) == diffed
 
+    def test_nan(self):
+        value = float('nan')
+        diffed = list(diff([value], [value]))
+        assert diffed == []
+
```
Comment (jirikuncar) [inline tests/test_dictdiffer.py]: please test also `float('nan')` and number

File: dictdiffer/__init__.py (hunk 1 above)
Comment (thehesiod) [inline dictdiffer/__init__.py]: that’s not comparing first to second.

File: tests/test_dictdiffer.py (hunk 2 above)
Comment (thehesiod) [inline tests/test_dictdiffer.py]: ensuring decimal.Decimal(‘nan’) behaves like float(‘nan’) is up to the python language, not really the focus of this test.

Comment (jirikuncar) [inline tests/test_dictdiffer.py]: I was more thinking about `diff([float('nan')], [3.5]))` just to have it covered.

**Comment by jirikuncar** [comment]
```diff
diff --git a/dictdiffer/__init__.py b/dictdiffer/__init__.py
index 7514e29..3b82ddb 100644
--- a/dictdiffer/__init__.py
+++ b/dictdiffer/__init__.py
@@ -197,7 +197,7 @@ def diff(first, second, node=None, ignore=None, path_limit=None, expand=False):
 
     else:
         # Compare string and integer types and yield `change` flag.
-        if first != second:
+        if first != second and not (first != first and second != second):
             yield CHANGE, dotted_node, (first, second)
 
 
diff --git a/tests/test_dictdiffer.py b/tests/test_dictdiffer.py
index f90d49c..5266f87 100644
--- a/tests/test_dictdiffer.py
+++ b/tests/test_dictdiffer.py
@@ -195,6 +195,11 @@ class DictDifferTests(unittest.TestCase):
         diffed = next(diff(first, second))
         assert ('change', 'a', (['a'], 'a')) == diffed
 
+    def test_nan(self):
+        value = float('nan')
+        diffed = list(diff([value], [value]))
+        assert diffed == []
+
     def test_unicode_keys(self):
         first = {u'привет': 1}
         second = {'hello': 1}
```

File: dictdiffer/__init__.py (hunk 1 above)
Comment (jirikuncar) [inline dictdiffer/__init__.py]: please see https://github.com/inveniosoftware/dictdiffer/pull/55#issuecomment-145604632

If you have any test case that it is not covered please let me know. Thanks

File: tests/test_dictdiffer.py (hunk 2 above)
Comment (thehesiod) [inline tests/test_dictdiffer.py]: done!

**Comment by jirikuncar** [comment]
@thehesiod what do you think about my previous proposal https://github.com/inveniosoftware/dictdiffer/pull/55#issuecomment-145604632 ? 

Can I ask you also to squash commit message and follow our commit message style based on `git log`?

- [ ] Please rebase on top of latest `master` branch (and keep it updated until your PR is merged).
- [ ] Please add yourself to `AUTHORS`.
- [ ] Please reply if you agree to donate this (and any future) contributions to *dictdiffer* package under the terms of the Revised BSD Licence.

Thank you!

**Comment by thehesiod** [comment]
as long as it works I don’t really care. I give you full permission to use any of my changes in any way you see fit.  Do you want to go ahead and just check in your patch?  I can withdraw my pull request.

**Pull request closed without merging**